examples/fun-token/src/lib.rs

```diff
@@ -59,8 +59,15 @@ pub struct FunToken {
     pub total_supply: Balance,
 }
 
-#[near_bindgen(init => new)]
+impl Default for FunToken {
+    fn default() -> Self {
+        panic!("Fun token should be initialized before usage")
+    }
+}
+
+#[near_bindgen]
 impl FunToken {
+    #[init]
     pub fn new(owner_id: AccountId, total_supply: Balance) -> Self {
```

Comment:
This seems like a duplicated code that most contract will have to implement if there is a constructor.
There are 2 issues:
1) Manually disabling `Default` using panic. I think before this, the `Default` was not required, so if you wanted to avoid default, you didn't have to implement it.
2) `init` doesn't prevent reinitializing again. You have to manually implement state checking. As part of the `ethbridge` I used the following line in the `init` wasm function
```
assert!(near_bindgen::env::state_read::<EthBridge>().is_none(), "Already initialized");
```

Reply:
We can later add feature `#[init_once]`.